src/rcms/fm/app/level1/HCALxmlHandler.java

```diff
@@ -223,6 +223,10 @@ public String stripExecXML(String execXMLstring, ParameterSet<FunctionManagerPar
       for (StringT maskedApp: maskedAppArray) {
         //logger.info("[JohnLogVector] " + functionManager.FMname + ": about to start masking " + maskedApp.getString());
         String[] maskedAppParts = maskedApp.getString().split("_");
+        if (maskedAppParts.length != 2 || HCALlevelOneEventHandler.stringToInt(maskedAppParts[0]) || !(HCALlevelOneEventHandler.stringToInt(maskedAppParts[1]))){
```

Comment:
The way stripExecXML uses the MASKED_RESOURCES list could be optimized, sure, but the way it works is fairly straightforward and working on optimizing it to sort out the relevant types of resources seems like an overkill effort to improve code that is invisible to the user and works fine except when it encounters broken user input in the run key.

Maybe the simplest solution for now is just to validate the input from the run key and leave stripExecXML alone.